astropy/io/fits/fitsrec.py

```diff
@@ -479,6 +479,23 @@ def __repr__(self):
         # one added for recarray in Numpy 1.10) for backwards compat
         return np.ndarray.__repr__(self)
 
+    def __getattribute__(self, attr):
```

Comment:
Hmm, at first I thought that is really weird, to try `object.__getattribute__` and if that fails, do something else -- this is exactly what `__getattr__` is for! But then, of course, I realized that since `np.recarray` overrides `__getattribute__`, you really have no choice but to override it as well. That's definitely silly in `np.recarray`...

Anyway, a minimal request would be to add a comment to that extent! But since I was playing with it anyway, I also thought it could be a bit shorter:
```
    def __getattribute__(self, attr):
        # Unfortunately, np.recarray overrides __getattribute__, which
        # means we have to override it too, to change its behaviour.
        # Since this is called before anything else, as the first step,
        # we *have* to check whether the attribute simply exists.
        # (Normally, one would use __getattr__ to avoid that.)
        try:
            return object.__getattribute__(self, attr)
        except AttributeError:
            # attr might still be a fieldname.  If we have column
            # definitions, we should access this via .field, as the
            # data may have to be scaled.
            if self._coldefs is not None and attr in self.columns.names:
                return self.field(attr)

        # If not, just let the usual np.recarray override deal with it.
        return super().__getattribute__(attr)
```

Reply:
Yes sorry, I should have explained this. I also need to call it first to keep the same behavior as Numpy, finding ndarray attributes first (so a `'var'` column will not be accessible with the dot notation, `np.var` is returned instead). But then we need to check for columns before calling `recarray.__getattribute__`.
Thanks for the shorter version, that's much better indeed!